pkg/controller/kubefedcluster/controller.go

```diff
@@ -98,43 +102,71 @@ func newClusterController(config *util.ControllerConfig, clusterHealthCheckConfi
 		&fedv1b1.KubeFedCluster{},
 		util.NoResyncPeriod,
 		&cache.ResourceEventHandlerFuncs{
-			DeleteFunc: cc.delFromClusterSet,
-			AddFunc:    cc.addToClusterSet,
+			DeleteFunc: func(obj interface{}) {
+				castObj, ok := obj.(*fedv1b1.KubeFedCluster)
+				if !ok {
```

Comment:
I appreciate your attention to detail here, but even this level of rigor won't preclude missing deletion events.  Would you be up for removing these changes from the PR in favor of a separate PR that used a finalizer to ensure reliable cleanup on deletion? That change seems orthogonal to eliminating updates in response to status changes.